linkerd/http/prom/src/body_data/metrics.rs

```diff
@@ -0,0 +1,94 @@
+//! Prometheus counters for request and response bodies.
+
+use linkerd_metrics::prom::{self, Counter, Family, Registry};
+
+/// Counters for response body frames.
+#[derive(Clone, Debug)]
+pub struct ResponseBodyFamilies<L> {
+    /// Counts the number of response body frames.
+    resp_body_frames_total: Family<L, Counter>,
+    /// Counts the total number of bytes in response body frames.
+    resp_body_frames_bytes: Family<L, Counter>,
+}
+
+/// Counters to instrument a request or response body.
+#[derive(Clone, Debug, Default)]
+pub struct BodyDataMetrics {
+    /// Counts the number of request body frames.
+    pub frames_total: Counter,
+    /// Counts the total number of bytes in request body frames.
+    pub frames_bytes: Counter,
+}
+
+// === impl ResponseBodyFamilies ===
+
+impl<L> Default for ResponseBodyFamilies<L>
+where
+    L: Clone + std::hash::Hash + Eq,
+{
+    fn default() -> Self {
+        Self {
+            resp_body_frames_total: Default::default(),
+            resp_body_frames_bytes: Default::default(),
+        }
+    }
+}
+
+impl<L> ResponseBodyFamilies<L>
+where
+    L: prom::encoding::EncodeLabelSet
+        + std::fmt::Debug
+        + std::hash::Hash
+        + Eq
+        + Clone
+        + Send
+        + Sync
+        + 'static,
+{
+    const RESP_BODY_FRAMES_TOTAL_NAME: &'static str = "resp_body_frames_total";
```

Comment:
We shouldn't introduce a new metric naming convetion (`resp_`). Existing metrics use the full `response_`.